### PR TITLE
Made the @OA\Server annotation be compatible with L5_SWAGGER_BASE_PATH environment variable

### PR DESCRIPTION
**Pull request by ymhuang0808**
Solve the issue #235. 
@DarkaOnLine Please help me review the PR. 
Thanks. :smile:

### CODE REVIEW

File: src/Generator.php [hunk 1]
```diff
@@ -139,7 +139,11 @@ protected function populateServers()
...
         if (config('l5-swagger.paths.base') !== null) {
             if ($this->isOpenApi()) {
-                $this->swagger->servers = [
+                if (! is_array($this->swagger->servers)) {
+                    $this->swagger->servers = [];
+                }
+
+                $this->swagger->servers[] = [
                     new \OpenApi\Annotations\Server(['url' => config('l5-swagger.paths.base')]),
                 ];
```
Comment (DarkaOnLine) [inline src/Generator.php]: I think it should be:

```php
$this->swagger->servers[] = new \OpenApi\Annotations\Server(['url' => config('l5-swagger.paths.base')]);
```

File: tests/GeneratorTest.php [hunk 2]
```diff
@@ -89,6 +89,11 @@ public function canGenerateApiJsonFileWithChangedBaseServer()
 
         $this->assertTrue(file_exists($this->jsonDocsFile()));
 
+        $this->get(route('l5-swagger.docs'))
+            ->assertSee('https://projects.dev/api/v1')
+            ->assertDontSee('basePath')
+            ->assertStatus(200);
+
```
Comment (DarkaOnLine) [inline tests/GeneratorTest.php]: Just append `->assertSee('https://projects.dev/api/v1')` after lines 98 and 103, cause now I assume we should see 2 servers in documentation files, yes?

Also, from where `https://projects.dev/api/v1'` URL is coming here?

File: src/Generator.php (hunk 1 above)
Comment (ymhuang0808) [reply src/Generator.php]: In the beginning, I think it should be like you mentioned. But, I found if there was not `@OA\Server` annotation, the value of `$this->swagger->servers` is `@OA\UNDEFINED` string. If `$this->swagger->servers` was appended, it show the following error:

```
1) Tests\GeneratorTest::canGenerateApiJsonFileWithChangedBaseServer
Error: [] operator not supported for strings
```

Comment (DarkaOnLine) [reply src/Generator.php]: But you have this `if (! is_array($this->swagger->servers))` which should owerrite string to array?

File: tests/GeneratorTest.php (hunk 2 above)
Comment (ymhuang0808) [reply tests/GeneratorTest.php]: Yes. The PR solved the overwritten the definition of `@OA\Server` annotations. Therefore, I think the testing should be checked if the URL values were seen or not.

Comment (ymhuang0808) [reply tests/GeneratorTest.php]: `https://projects.dev/api/v1` exists in [the line of code](https://github.com/DarkaOnLine/L5-Swagger/blob/master/tests/storage/annotations/OpenApi/Anotations.php#L25)

File: src/Generator.php (hunk 1 above)
Comment (ymhuang0808) [reply src/Generator.php]: Yes. in the line 142, it changes string to array. If I misunderstand your concerns, please let me know.

Comment (DarkaOnLine) [reply src/Generator.php]: if we do like this (like in your PR):
```php
$this->swagger->servers[] = [new \OpenApi\Annotations\Server(['url' => config('l5-swagger.paths.base')])];
```
The result of `$this->swagger->servers` will be:
```php
[
   //Custom server1,
   //Custom server2,
   [
      new \OpenApi\Annotations\Server(['url' => config('l5-swagger.paths.base')])
   ]
]
```

But previously it was:
```php
[
   new \OpenApi\Annotations\Server(['url' => config('l5-swagger.paths.base')])
]
```

IMHO final array should look like:
```php
[
   //Custom server1,
   //Custom server2,
   new \OpenApi\Annotations\Server(['url' => config('l5-swagger.paths.base')])
]
```

Comment (ymhuang0808) [reply src/Generator.php]: @DarkaOnLine Sorry, I had not found the `\OpenApi\Annotations\Server` object exists in an array. `$this->swagger-servers` should be: 

```
[
   //Custom server1,
   //Custom server2,
   new \OpenApi\Annotations\Server(['url' => config('l5-swagger.paths.base')])
]
```

And it's appropriate.

I think I was dizzy in afternoon. :persevere:

File: tests/GeneratorTest.php [hunk 3]
```diff
@@ -91,6 +91,7 @@ public function canGenerateApiJsonFileWithChangedBaseServer()
 
         $this->get(route('l5-swagger.docs'))
             ->assertSee('https://test-server.url')
+            ->assertSee('https://projects.dev/api/v1')
```
[the remark this answers is not in the document]
Comment (ymhuang0808) [reply tests/GeneratorTest.php]: I forgot that the JSON file should be asserted.

**Comment by ymhuang0808** [comment]
@DarkaOnLine I made several commits for this PR. May I kindly ask for squashing these commits into a single commit when you would like to merge the PR?

**Comment by DarkaOnLine** [comment]
@ymhuang0808 - thanks for your input into this. Great job!

**Pull request merged**